src/main/java/org/repodriller/RepoVisitor.java

```diff
@@ -101,41 +157,51 @@ void beginRepoVisit(SCMRepository repo) {
 	 * @param commit The commit that will be visited
 	 */
 	void visitCommit(Commit commit) {
-		for(Map.Entry<CommitVisitor, PersistenceMechanism> entry : visitors.entrySet()) {
-			CommitVisitor visitor = entry.getKey();
-			PersistenceMechanism writer = entry.getValue();
+		log.debug("Visiting commit " + commit.getHash());
+		/* Get a clone from the pool. */
+		SCMRepository scmRepoClone = getSCMRepositoryClone();
 
+		/* Have each visitor process this commit. */
+		for (CVPM cvpm : visitors) {
 			try {
-				log.info("-> Processing " + commit.getHash() + " with " + visitor.name());
-				visitor.process(currentRepo, commit, writer);
+				log.debug("Thread " + Thread.currentThread().getId() + ": processing " + commit.getHash() + " with " + cvpm.cv.name() + " in clone " + scmRepoClone.getPath());
+				cvpm.cv.process(scmRepoClone, commit, cvpm.pm);
 			} catch (CSVFileFormatException e) {
 				log.fatal(e);
 				System.exit(-1);
 			} catch (Exception e) {
-				log.error("error processing #" + commit.getHash() + " in " + currentRepo.getPath() +
-						", processor=" + visitor.name() + ", error=" + e.getMessage(), e);
+				log.error("Error processing #" + commit.getHash() + " in clone " + scmRepoClone.getPath() +
```

Comment:
If this might throw an exception, we will leak the scmRepoClone. Should wrap the loop with a try-finally.